src/server_manager/web_app/gcp_account.ts

```diff
@@ -252,28 +263,41 @@ export class GcpAccount implements gcp.Account {
     };
     const createInstanceOperation =
         await this.apiClient.createInstance(zoneLocator, createInstanceData);
-    if (createInstanceOperation.error?.errors) {
-      // TODO: Throw error.
+    const errors = createInstanceOperation.error?.errors;
+    if (errors) {
+      throw new ServerInstallFailedError(`Instance creation failed: ${errors}`);
     }
 
-    const instanceLocator = {instanceId: createInstanceOperation.targetId, ...zoneLocator};
-    const instance = await this.apiClient.getInstance(instanceLocator);
+    const instanceId = createInstanceOperation.targetId;
+    const instanceCreation = this.apiClient.computeEngineOperationZoneWait(zoneLocator, createInstanceOperation.name).then(async () => {
+      const instanceLocator = {instanceId, ...zoneLocator};
+      try {
+        await this.promoteEphemeralIp(instanceLocator);
+      } catch (e) {
+        await this.apiClient.deleteInstance(instanceLocator);
```

Comment:
I'm questioning whether we should delete the instance. We could as well show the server as broken, so the user can manually delete it or perhaps fix it / try again.

Reply:
OK, this no longer deletes the instance.  Instead, `GcpHost.delete()` now attempts to delete servers that failed setup, and tolerates 404s on resources that it is trying to delete.